dex/farm/src/farm_token_merge.rs

```diff
@@ -104,29 +101,6 @@ pub trait FarmTokenMergeModule:
         self.weighted_average_ceil(dataset)
```

Comment:
The dataset vector would be created even if `weighted_average_ceil` was inlined. The Rust compiler might be smart enough to not create a native Vec<T> and optimize, but I don't think it can figure it out for a custom type like ManagedVec.